src/service.js

```diff
@@ -205,33 +251,41 @@ module.exports = function(mixinOptions) {
 							}
 
 							if (dataLoaderKey == null) {
-								return null;
+								result = null;
+							} else {
+								result = Array.isArray(dataLoaderKey)
+									? await dataLoader.loadMany(dataLoaderKey)
+									: await dataLoader.load(dataLoaderKey);
 							}
-
-							return Array.isArray(dataLoaderKey)
-								? await dataLoader.loadMany(dataLoaderKey)
-								: await dataLoader.load(dataLoaderKey);
 						} else if (fileUploadArg != null && args[fileUploadArg] != null) {
 							if (Array.isArray(args[fileUploadArg])) {
-								return await Promise.all(
+								result = await Promise.all(
 									args[fileUploadArg].map(async uploadPromise => {
 										const {
 											createReadStream,
 											...$fileInfo
 										} = await uploadPromise;
 										const stream = createReadStream();
-										return context.ctx.call(actionName, stream, {
-											meta: { $fileInfo },
-										});
+										return context.ctx.call(
+											actionName,
+											stream,
+											{ meta: { $fileInfo } },
+											await this.actions.actionOptions(root, args, context)
```

Comment:
Can this pattern be changed to only have the call to `this.actions.actionOptions` a single time and assign its result to a variable that is used for all of the calls?

One other note on these options is it seems like it would need to apply to the DataLoader callback as well, so the options should probably be passed to that function and added to the call there.

Reply:
So... I'm trying to make the change mentioned... but for some reason, calling `this.actions.actionOptions` earlier is magically adding `{timeout: 0}` to any existing params. The only change is me moving the call earlier. Does this make sense to anyone?